packages/doenetml-worker-rust/lib-doenetml-core/src/utils/keyvalue.rs

```diff
@@ -19,4 +20,18 @@ where
 
         None
     }
+
+    /// Match key to the HashMap keys, ignoring case.
+    /// If found, remove the original value from the HashMap.
+    fn remove_ignore_case(&mut self, key: &str) -> Option<V> {
+        let mut found_key_option = None;
```

Comment:
Functional style would look nicer.

```rust
self.keys().find(|&k| k.to_string().eq_ignore_ascii_case(key).and_then(...)
```

Reply:
I tried approaches like this first, but then it got mad at me because `k` is borrowed and `remove` wouldn't work. I finally gave up and did this two-step approach.

Reply:
A clone was missing.
```
        self.keys()
            .find(|k| k.to_string().eq_ignore_ascii_case(key))
            .map(|k| k.clone())
            .and_then(|k| self.remove(&k))
```
works, but we might consider using `.retain`.